code/mob/living/critter/flock/flockdrone.dm

```diff
@@ -210,32 +197,20 @@
 	if (!src.flock)
 		..()
 		return
-
 	src.flock.hideAnnotations(src)
-
+	//hold a ref to the controller so we can put them back with the flock
+	var/mob/living/intangible/flock/old_controller = src.controller
 	if (src.controller)
+		src.release_control_abrupt(FALSE)
+	if (old_controller)
```

Comment:
```suggestion
```

Reply:
that would runtime if you call dormantize on a drone with no controller

Reply:
Yeah, but if there's old_controller, there's also a controller, since old_controller was assigned to controller?

Reply:
correct, but we don't know that there was a controller